src/Workspaces/Core/Portable/Formatting/FormattingOptions.cs

```diff
@@ -99,6 +99,9 @@ private static string GetEndOfLineEditorConfigString(string option)
 
         internal static Option<bool> AllowDisjointSpanMerging { get; } = CreateOption(OptionGroup.Default, nameof(AllowDisjointSpanMerging), defaultValue: false);
 
+        internal static readonly PerLanguageOption<bool> AutoFormattingOnReturn = CreatePerLanguageOption(OptionGroup.Default, nameof(AutoFormattingOnReturn), defaultValue: true,
+            storageLocations: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.Auto Formatting On Return"));
```

Comment:
this option location is very specific to VS. so I am not sure what it means that Workspace layer saving option to place where it is specific to VS.

Reply:
the option itself is a bit strange.  it's used in one place:

```c#
        public static bool ShouldUseSmartTokenFormatterInsteadOfIndenter(
            IEnumerable<AbstractFormattingRule> formattingRules,
            CompilationUnitSyntax root,
            TextLine line,
            OptionSet optionSet,
            CancellationToken cancellationToken)
        {
            Contract.ThrowIfNull(formattingRules);
            Contract.ThrowIfNull(root);

            if (!optionSet.GetOption(FormattingOptions.AutoFormattingOnReturn, LanguageNames.CSharp))
            {
                return false;
            }
```

Which is in the indentation service itself.  I think it's a way to allow us (potentially for manual testing purposes) to opt out of smart-indent.  It's a bit wonky, but i'm not sure to do about it.  We could try to figure something else out once the following two PRs go in here.